src/main/java/io/openshift/launchpad/ui/osio/NewProjectWizard.java

```diff
@@ -0,0 +1,221 @@
+   @Inject
+   private BoosterCatalogServiceFactory catalogServiceFactory;
+   @Inject
+   private ProjectFactory projectFactory;
+   @Inject
+   private MavenBuildSystem mavenBuildSystem;
+
+   @Override
+   public void initializeUI(UIBuilder builder) throws Exception
+   {
+      List<BoosterDTO> boosterList = new ArrayList<>();
+      boosters.clear();
+      UIContext uiContext = builder.getUIContext();
+      BoosterCatalogService catalogService = getCatalogService(uiContext);
+      if (catalogService == null)
+      {
+         throw new IllegalArgumentException("No BoosterCatalogService");
+      }
+      Set<Mission> missions = catalogService.getMissions();
+      for (Mission m : missions)
+      {
+         Set<Runtime> runtimes = catalogService.getRuntimes(m);
+         for (Runtime r : runtimes)
+         {
+            Optional<Booster> optional = catalogService.getBooster(m, r);
+            if (optional.isPresent())
+            {
+               Booster b = optional.get();
+               boosters.add(b);
+               boosterList.add(new BoosterDTO(b));
+            }
+         }
+      }
+      booster.setValueChoices(boosterList);
+      booster.setItemLabelConverter(booster -> booster.getName());
+      builder.add(booster).add(named).add(topLevelPackage);
+      //.add(version);
+   }
+
+   @Override
+   public UICommandMetadata getMetadata(UIContext context)
+   {
+      return Metadata.forCommand(getClass()).name("Launchpad: New Project")
+               .description("Generate your project from a booster")
+               .category(Categories.create("Openshift.io"));
+   }
+
+   @Override
+   public void validate(UIValidationContext context)
+   {
+   }
+
+   @Override
+   public NavigationResult next(UINavigationContext context) throws Exception
+   {
+      Map<Object, Object> attributeMap = context.getUIContext().getAttributeMap();
+      attributeMap.put("name", named.getValue());
+      attributeMap.put("type", getTypeValue());
+      return null;
+   }
+
+   @Override
+   public Result execute(UIExecutionContext context) throws Exception
+   {
+      BoosterDTO boosterDTO = getTypeValue();
+      Booster booster = null;
+      String boosterId = null;
+      if (boosterDTO != null)
+      {
+         boosterId = boosterDTO.getId();
+         for (Booster b : boosters)
+         {
+            if (boosterId.equals(b.getId()))
+            {
+               booster = b;
+               break;
+            }
+         }
+      }
+      if (booster == null)
+      {
+         return Results.fail("Could not find boister " + boosterId);
+      }
+      DirectoryResource initialDir = (DirectoryResource) context.getUIContext().getInitialSelection().get();
+      DirectoryResource projectDirectory = initialDir.getChildDirectory(named.getValue());
+      // Using ProjectFactory to invoke bound listeners
+      Project project = projectFactory.createProject(projectDirectory, mavenBuildSystem);
```

Comment:
No need to use the project API here. The BoosterCatalogService.copy method already creates the needed structure